WapoJS/src/host_functions.rs

```diff
@@ -86,9 +86,13 @@ pub(crate) fn setup_host_functions(ctx: &js::Context, cfg: &ServiceConfig) -> Re
         #[cfg(feature = "wapo")]
         query_listen::setup(&ns)?;
         wapo_ocalls::setup(&ns)?;
+        #[cfg(not(feature = "native"))]
         derive_secret::setup(&ns)?;
     }
 
+    #[cfg(feature = "native")]
+    derive_secret::setup(&ns)?;
```

Comment:
So, if running in wasm, the isolateEval wouldn't get a `deriveSecret`. Which means the user codes works in native build won't work in real SGX workers.
Actually, as there is no cross worker deterministic state at all, it even hard to implement the `deriveSecret` in the future for the worker.

Do you think it's better to implement and polyfill the `Pink.deriveSecret` in pure JS? like following steps:
- Get the root secret from external server at startup.
- Derive secret for user script before calling `isolateEval`.
- Add polyfill scripts for the `isolateEval`.

Reply:
> So, if running in wasm, the isolateEval wouldn't get a deriveSecret. Which means the user codes works in native build won't work in real SGX workers.
I removed the `#[cfg(feature = "native")]` flag because I was unsure how to implement the secret acquire workflow for non-native builds and how to test it. To address this, I added the flag as a "feature flag" specifically for testing purposes, ensuring that it does not introduce any new behavior in the wasm build.

Reply:
> I was unsure how to implement the secret acquire workflow for non-native builds

As i sad, it's hard to implement such feature in the engine, because the worker is stateless.

So, I think it might be better to implement the `deriveSecret` in pure JS at the workload, because the workload has more flexible ways to get the root secret(e.g. from a kms server, from env).